src/util-inl.h

```diff
@@ -229,30 +230,85 @@ bool StringEqualNoCaseN(const char* a, const char* b, size_t length) {
   return true;
 }
 
+// clang
+#ifndef __has_builtin
+#define __has_builtin(x) (0)
+#endif
+
+template<typename T>
```

Comment:
Tiny style nit but in most places we write `template <typename T>` (space before the `<`.)